### PR TITLE
Optimizing E2E.

### PR DESCRIPTION
**Pull request by iamhucong**
- Optimizing E2E logs.
- Fix the issue of encountering an exception during E2E data filling without generating a log.
- Fix the issue of the database container not being fully initialized when starting e2e tests.
- Fix the issue where the cast type in E2E generateInsertSQL is incorrect.
- Fix the issue of sudden container shutdown during e2e execution.
- Fix the issue where the assert time is not formatted correctly. 
- Fix the problem of the t_product ddl not being present in the PostgreSQL tbl E2e.

---

Before committing this PR, I'm sure that I have checked the following options:
- [ ] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [ ] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [ ] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added corresponding unit tests for my changes.

### CODE REVIEW

File: test/e2e/env/src/test/java/org/apache/shardingsphere/test/e2e/env/container/atomic/storage/DockerStorageContainer.java [hunk 1]
```diff
@@ -76,11 +80,12 @@ protected void configure() {
             withClasspathResourceMapping(new ScenarioDataPath(scenario).getInitSQLResourcePath(Type.EXPECTED, databaseType) + "/01-expected-init.sql",
                     "/docker-entrypoint-initdb.d/01-expected-init.sql", BindMode.READ_ONLY);
         }
+        withClasspathResourceMapping("/container/init-sql/" + databaseType.getType().toLowerCase() + "/99-be-ready.sql", "/docker-entrypoint-initdb.d/99-be-ready.sql", BindMode.READ_ONLY);
```
Comment (strongduanmu) [inline test/e2e/env/src/test/java/org/apache/shardingsphere/test/e2e/env/container/atomic/storage/DockerStorageContainer.java:83]: Why add 99-be-ready.sql?

File: test/e2e/sql/src/test/java/org/apache/shardingsphere/test/e2e/framework/param/model/CaseTestParameter.java [hunk 2]
```diff
@@ -44,6 +44,8 @@ public final class CaseTestParameter implements E2ETestParameter {
     
     @Override
     public String toString() {
-        return String.format("%s: %s -> %s -> %s", adapter, scenario, databaseType.getType(), testCaseContext.getTestCase().getSql());
+        final String sql = null == testCaseContext ? null : testCaseContext.getTestCase().getSql();
```
Comment (strongduanmu) [inline test/e2e/sql/src/test/java/org/apache/shardingsphere/test/e2e/framework/param/model/CaseTestParameter.java:47]: Please remove final modifier here.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: test/e2e/env/src/test/java/org/apache/shardingsphere/test/e2e/env/container/atomic/storage/DockerStorageContainer.java (hunk 1 above)
Comment (iamhucong) [reply test/e2e/env/src/test/java/org/apache/shardingsphere/test/e2e/env/container/atomic/storage/DockerStorageContainer.java:83]: This is to create a 'ready' user, which is used to indicate to the code that the initialization scripts have all been executed.

File: test/e2e/sql/src/test/java/org/apache/shardingsphere/test/e2e/framework/param/model/CaseTestParameter.java (hunk 2 above)
Comment (iamhucong) [reply test/e2e/sql/src/test/java/org/apache/shardingsphere/test/e2e/framework/param/model/CaseTestParameter.java]: Okay.

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**